src/transformers/models/llama/modeling_llama.py

```diff
@@ -149,6 +149,140 @@ def forward(self, x, position_ids):
         return cos, sin
 
 
+class LlamaYarnScalingRotaryEmbedding(LlamaRotaryEmbedding):
+    def __init__(
+        self,
+        dim,
+        max_position_embeddings=2048,
+        base=10000,
+        scaling_factor=1,
+        original_max_position_embeddings=2048,
+        attention_factor=None,
+        beta_fast=32,
+        beta_slow=1,
+        device=None,
+    ):
+        super().__init__(dim, max_position_embeddings, base, device, scaling_factor)
+
+        self.original_max_position_embeddings = original_max_position_embeddings
+        self.attention_factor = attention_factor
+        self.beta_fast = beta_fast
+        self.beta_slow = beta_slow
+
+        if self.attention_factor is None:
+            self.attention_factor = 0.1 * math.log(scaling_factor) + 1.0
```

Comment:
Let's add a note saying that this is the default value according to the yarn paper, so it doesn't look like a magic number :)